subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/install/internal/AdoptOpenJdkRemoteBinary.java

```diff
@@ -0,0 +1,126 @@
+    private final AdoptOpenJdkDownloader downloader;
+
+    @Inject
+    public AdoptOpenJdkRemoteBinary(SystemInfo systemInfo, OperatingSystem operatingSystem, AdoptOpenJdkDownloader downloader) {
+        this.systemInfo = systemInfo;
+        this.operatingSystem = operatingSystem;
+        this.downloader = downloader;
+    }
+
+    // TODO: [bm] this might be downloading the same jdk multiple times in parallel builds
+    public Optional<File> download(JavaToolchainSpec spec) {
+        if (!canProvidesMatchingJdk(spec)) {
+            return Optional.empty();
+        }
+        File tmpFile = new File(Files.createTempDir(), toFilename(spec));
+        URI source = toDownloadUri(spec);
+        downloader.download(source, tmpFile);
+        return Optional.of(tmpFile);
+    }
+
+    private boolean canProvidesMatchingJdk(JavaToolchainSpec spec) {
+        return getLanguageVersion(spec).isJava8Compatible();
+    }
+
+    URI toDownloadUri(JavaToolchainSpec spec) {
+        return constructUri(spec);
+    }
+
+    private URI constructUri(JavaToolchainSpec spec) {
+        return URI.create(getServerBaseUri() +
+            "v3/binary/latest/" + getLanguageVersion(spec).getMajorVersion() +
+            "/" +
+            determineReleaseState() +
+            "/" +
+            determineOs() +
+            "/" +
+            determineArch() +
+            "/jdk/hotspot/normal/adoptopenjdk");
+    }
+
+    String toFilename(JavaToolchainSpec spec) {
+        return String.format("adoptopenjdk-%s-%s-%s.%s", getLanguageVersion(spec), determineArch(), determineOs(), determineFileExtension());
+    }
+
+    private String determineFileExtension() {
+        if (operatingSystem.isWindows()) {
+            return "zip";
+        }
+        return "tar.gz";
+    }
+
+    private JavaVersion getLanguageVersion(JavaToolchainSpec spec) {
+        return spec.getLanguageVersion().get();
+    }
+
+    private String determineArch() {
+        switch (systemInfo.getArchitecture()) {
+            case i386:
+                return "x32";
+            case amd64:
+                return "x64";
+            case aarch64:
+                return "aarch64";
+        }
+        return systemInfo.getArchitectureName();
+    }
+
+    private String determineOs() {
+        if (operatingSystem.isWindows()) {
+            return "windows";
+        } else if (operatingSystem.isMacOsX()) {
+            return "mac";
+        } else if (operatingSystem.isLinux()) {
+            return "linux";
+        }
+        return operatingSystem.getFamilyName();
+    }
+
+    // TODO: [bm] determine using API?
+    // https://github.com/AdoptOpenJDK/openjdk-api-v3/issues/256
+    private String determineReleaseState() {
+        return "ga";
+    }
+
+    private String getServerBaseUri() {
+        String baseUri = System.getProperty("org.gradle.jvm.toolchain.install.internal.adoptopenjdk.baseUri", "https://api.adoptopenjdk.net/");
```

Comment:
We need to be able to configure this. The open jdk api should work just fine behind a caching proxy such as Nexus or Artifactory.

Reply:
I'm not sure exactly what you refer to with "this". What do you need to be configurable? The base URL?

Reply:
Apologies, typing from phone. 

Yes it would be very useful for users in companies where downloading binaries directly from Internet is against policy to be able to configure the base URL of the Open Jdk API. 

The current configuration is marked as internal, which means that by policy we will not be able to use it for production setup.

Reply:
Yes, it will likely stay internal for the first release (similar to APIs being `@Incubating`). Nevertheless, downloading the toolchain is a last resort, you can still use toolchains using locally available JDKs if downloading is not something that is possible in your environment.